ethereum/tests/test_contracts_solidity.py

```diff
@@ -0,0 +1,1653 @@
+import os
+from os import path
+import binascii
+import pytest
+import bitcoin
+from secp256k1 import PrivateKey
+
+
+from ethereum import tester
+from ethereum import abi
+from ethereum import utils
+from ethereum import _solidity
+from ethereum._solidity import get_solidity
+
+solidity = get_solidity()
+
+SOLIDITY_AVAILABLE = get_solidity() is not None
+
+
+@pytest.mark.skipif(not SOLIDITY_AVAILABLE, reason='solc compiler not available')
```

Comment:
It might be nice to roll this up into something reusable so that it isn't repeated for every test in the file.


```python
skip_if_no_solidity_compiler = pytest.mark.skipif(not SOLIDITY_AVAILABLE, reason='solc compiler not available')

@skip_if_no_solidity_compiler
def test_things():
     ...
```